File: code_samples/haskell/database_query_builder.hs
-- Database: Query Builder
-- AI/ML Training Sample

module QueryBuilder where

data QueryBuilder = QueryBuilder {
    getData :: String
} deriving (Show, Eq)

process :: QueryBuilder -> String -> QueryBuilder
process obj input = obj { getData = input }

validate :: QueryBuilder -> Bool
validate obj = not (null (getData obj))

main :: IO ()
main = do
    let instance = QueryBuilder { getData = "" }
    let updated = process instance "example"
    putStrLn $ "Data: " ++ getData updated
    putStrLn $ "Valid: " ++ show (validate updated)

File: code_samples/go/database_query_builder.go
package main

import (
    "fmt"
)

// Database: Query Builder
// AI/ML Training Sample

type QueryBuilder struct {
    Data string
}

func NewQueryBuilder() *QueryBuilder {
    return &QueryBuilder{
        Data: "",
    }
}

func (s *QueryBuilder) Process(input string) {
    s.Data = input
}

func (s *QueryBuilder) Validate() bool {
    return len(s.Data) > 0
}

func (s *QueryBuilder) GetData() string {
    return s.Data
}

func main() {
    instance := NewQueryBuilder()
    instance.Process("example")
    fmt.Printf("Data: %s\n", instance.GetData())
    fmt.Printf("Valid: %v\n", instance.Validate())
}
